Core/Src/main.c

```diff
@@ -87,16 +90,24 @@ int main(void)
   /* Initialize all configured peripherals */
   MX_GPIO_Init();
   MX_USART2_UART_Init();
+  MX_SPI1_Init();
+  MX_TIM1_Init();
   /* USER CODE BEGIN 2 */
 
+  HAL_NVIC_SetPriority(TIM1_CC_IRQn, 0, 0);
+  HAL_NVIC_EnableIRQ(TIM1_CC_IRQn);
+
+  TIM1->ARR = 64000; // Set period
+  TIM1->CR1 |= TIM_CR1_CEN; // Set the enable bit to 1, start the timer
```

Comment:
Hmmm if you had to do this manually, did you still do it in the STM32 CUBE IDE? In general, it is always best to use the IDE to generate this type of code although this will still work.